File: ProjectTrackercs/www/PTWebService/App_Code/ProjectData.cs
using System;
using System.Collections.Generic;

public class ProjectData
{
  private Guid _id;
  private string _name;
  private string _started;
  private string _ended;
  private string _description;
  private List<ProjectResourceData> 
    _resources = new List<ProjectResourceData>();

  public Guid Id
  {
    get { return _id; }
    set { _id = value; }
  }

  public string Name
  {
    get { return _name; }
    set { _name = value; }
  }

  public string Started
  {
    get { return _started; }
    set { _started = value; }
  }

  public string Ended
  {
    get { return _ended; }
    set { _ended = value; }
  }

  public string Description
  {
    get { return _description; }
    set { _description = value; }
  }

  public void AddResource(ProjectResourceData resource)
  {
    _resources.Add(resource);
  }

  public ProjectResourceData[] ProjectResources
  {
    get
    {
      if (_resources.Count > 0)
        return _resources.ToArray();
      return null;
    }
    set
    {
      _resources = new List<ProjectResourceData>(value);
    }
  }
}

File: samples/CslaNet/vb/ProjectTrackervb/www/PTWcfService/App_Code/ProjectData.vb
Imports System.Collections.Generic
Imports Microsoft.VisualBasic
Imports System.Runtime.Serialization

<DataContract()> _
Public Class ProjectData

  Private _id As Guid
  Private _name As String
  Private _started As String
  Private _ended As String
  Private _description As String
  Private _resources As List(Of ProjectResourceData) = New List(Of ProjectResourceData)()

  <DataMember()> _
  Public Property Id() As Guid
    Get
      Return _id
    End Get
    Set(ByVal value As Guid)
      _id = value
    End Set
  End Property

  <DataMember()> _
  Public Property Name() As String
    Get
      Return _name
    End Get
    Set(ByVal value As String)
      _name = value
    End Set
  End Property

  <DataMember()> _
  Public Property Started() As String
    Get
      Return _started
    End Get
    Set(ByVal value As String)
      _started = value
    End Set
  End Property

  <DataMember()> _
  Public Property Ended() As String
    Get
      Return _ended
    End Get
    Set(ByVal value As String)
      _ended = value
    End Set
  End Property

  <DataMember()> _
  Public Property Description() As String
    Get
      Return _description
    End Get
    Set(ByVal value As String)
      _description = value
    End Set
  End Property

  Public Sub AddResource(ByVal resource As ProjectResourceData)
    _resources.Add(resource)
  End Sub

  <DataMember()> _
  Public Property ProjectResources() As ProjectResourceData()
    Get
      If _resources.Count > 0 Then
        Return _resources.ToArray()
      End If
      Return Nothing
    End Get
    Set(ByVal value As ProjectResourceData())
      _resources = New List(Of ProjectResourceData)(value)
    End Set
  End Property

End Class
